src/test/groovy/com/vacation_bot/vacation/VacationIntegrationTest.groovy

```diff
@@ -0,0 +1,160 @@
+
+        when: 'the controller is called'
+        String result = getResponse(inputData).getBody()
+
+        then: 'the result is correct'
+        result == expectedResult
+        def createdVacationTotal = vacationTotalRepository.findAll().stream().filter { e -> e.userId == userId && e.year == vacationTotalYear }.findFirst().get()
+        createdVacationTotal
+        createdVacationTotal.vacationTotal == expectedVacationTotalDays
+
+        def createdVacation = vacationModelRepository.findAll().stream().filter { e -> e.userId == userId }.findFirst()
+        def createdVacationPresent = createdVacation.isPresent()
+        createdVacationPresent == expectedCreatedVacation
+//        with(createdVacationPresent) {
+//            LocalDateTime.ofInstant(Instant.ofEpochMilli(createdVacation.get().startDate), ZoneOffset.UTC).toLocalDate() == parsedStartDate
+//            LocalDateTime.ofInstant(Instant.ofEpochMilli(createdVacation.get().endDate), ZoneOffset.UTC).toLocalDate() == LocalDate.parse(newEndDate)
+//        }
+
+        where:
+        newEndDate      ||  expectedCreatedVacation     ||  expectedVacationTotalDays                                                                           ||  expectedResult
+        '2017-10-15'    ||  true                        ||  defaultVacationTotalDays - ChronoUnit.DAYS.between(parsedStartDate, LocalDate.parse(newEndDate))    ||  "The registration of your vacation from ${startDate} to ${newEndDate} was successfully completed! You have left ${defaultVacationTotalDays - ChronoUnit.DAYS.between(parsedStartDate, LocalDate.parse(newEndDate))} days"
+        '2017-10-22'    ||  true                        ||  defaultVacationTotalDays - ChronoUnit.DAYS.between(parsedStartDate, LocalDate.parse(newEndDate))    ||  "The registration of your vacation from ${startDate} to ${newEndDate} was successfully completed! You don't have vacation days"
+        '2017-10-25'    ||  false                       ||  20                                                                                                  ||  "You can not receive vacation. You have ${availableDays} days."
+    }
+
+    def 'exercise controller (vacation total is not null)'() {
+
+        given: 'the vacation total is saved to the db'
+        validVacationTotal.vacationTotal = newVacationTotalDays
+        vacationTotalRepository.save(validVacationTotal)
+
+        and: 'the user is saved to the db'
+        userRepository.save(validUser)
+
+        and: 'set new end date'
+        inputData.endDate = newEndDate
+
+        when: 'the controller is called'
+        String result = getResponse(inputData).getBody()
+
+        then: 'result is correct'
+        result == expectedResult
+        validVacationTotal.vacationTotal == newVacationTotalDays
+        vacationModelRepository.findAll().isEmpty() == expectedVacationList
+
+        where:
+        newEndDate      ||  newVacationTotalDays    ||  expectedVacationList    ||  expectedResult
```

Comment:
`|` is used between each variable, `||` is used between input values on the left side and expected results on the right. 
`newEndDate  | newVacationTotalDays || expectedVacationList | expectedResult`